### PR TITLE
keycloak_client: remove code that turns attributes dict into list

### PR DESCRIPTION
**Pull request by fgruenbauer**
##### SUMMARY
The `attribute` config item contains a dict with additional settings. The module turns the dict into a list before storing it for the diff/end_state, which turns it into a list of just the keys:

```
+    "attributes": [
+        "backchannel.logout.revoke.offline.tokens",
+        "backchannel.logout.session.required",
+        "frontchannel.logout.url"
+    ],
+    "baseUrl": "https://www.example.com/",
```

The actual data sent to the API is not changed, so only `--diff`/`end_state` is affected.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

keycloak_client
##### ADDITIONAL INFORMATION

Create a client with the module with any of the `attributes` settings from above set and `--diff`.

### CODE REVIEW

File: plugins/modules/keycloak_client.py [hunk 1]
```diff
@@ -805,9 +805,6 @@ def normalise_cr(clientrep, remove_ids=False):
     # Avoid the dict passed in to be modified
     clientrep = clientrep.copy()
 
-    if 'attributes' in clientrep:
-        clientrep['attributes'] = list(sorted(clientrep['attributes']))
```
Comment (felixfontein) [inline plugins/modules/keycloak_client.py:809]: Is it possible that this used to be a list, and was changed to a dict at some point?

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: plugins/modules/keycloak_client.py (hunk 1 above)
Comment (fgruenbauer) [reply plugins/modules/keycloak_client.py:809]: It kinda looks like it. But i checked Keycloak 5, 9, 13, 18, 25 and its always a dict. Same for the clientscopes.

Comment (felixfontein) [reply plugins/modules/keycloak_client.py:809]: This was added in #3610 by @pmdumuid. @pmdumuid do you remember why you chose this? The tests added in the PR use Keycloak 12.0.4, maybe that version uses it for whatever reason (even if 9 and 13 do not)?

**Comment by pmdumuid** [comment]
Hey Felix,

I added it because when  I ran ansible in check mode, I found that
sometimes (possibly after a keycloak restart) the order of attribute didn't
match between desired and what keycloak returned.  Thus a false difference
is reported because the order of attributes doesn't matter I chose to add
the sorting there before determining if there is a difference.

I think I have might have more changes like this in my personal fork
because my use case was to deploy every few weeks, and only proceed if the
check passed when running the previous ansible code before the new ansible
code.

Note : answering in my phone without looking at the full diff.

**Comment by pmdumuid** [comment]
Sorry I have just reread what I responded wirh after sending it.  Please
replace

"I chose to"

with

".  I avoided this false difference in check mode by choosing to "

**Comment by felixfontein** [comment]
I decided to merge this since currently it seems that the PR is correct, and so that it can go into the 10.0.1 release. Thanks everyone!

**Pull request merged**